shopfloor/tests/test_actions_data.py

```diff
@@ -52,19 +52,38 @@ def setUpClassBaseData(cls):
         cls.picking.action_assign()
 
         cls.supplier = cls.env["res.partner"].sudo().create({"name": "Supplier"})
-        cls.vendor = (
+        cls.product_a_vendor = (
             cls.env["product.supplierinfo"]
             .sudo()
             .create(
                 {
                     "name": cls.supplier.id,
                     "price": 8.0,
-                    "product_code": "VENDOR_CODE",
+                    "product_code": "VENDOR_CODE_A",
                     "product_id": cls.product_a.id,
                     "product_tmpl_id": cls.product_a.product_tmpl_id.id,
                 }
             )
         )
+        cls.product_a_variant = cls.product_a.copy(
+            {
+                "name": "Product A variant 1",
+                "type": "product",
+                "default_code": "A-VARIANT",
+                "barcode": "A-VARIANT",
+            }
+        )
```

Comment:
Can you try to do a call to `flush()` right after this seller creation? I suspect it'll fail because the variants have the same attributes combination there. If it's the case, this issue could raise later if another module  in this repo perform this `flush`.

Reply:
@sebalix done, no fail :wink:

Reply:
Weird :) A similar issue happens on an OCA PR (don't remember which one). Anyway, OK!